### DISCUSSION TITLE
Ability to "lock" difficulty adjust for certain CS/OD/HP/AR

### DISCUSSION THREAD

**Discussion by adriankiezik** [ppy/osu#28476] [Ideas]
**Example:**
Player wants to practice maps with slighly lower approach rate

**Currently:**
When player switches map difficulty adjust does not consider map AR. (same AR is set for each map).
This forces player to customize difficulty adjust for each map, which can be tedious.

**Feature request:**
Add ability to lock certain CS/OD/HP/AR in difficulty adjust. If player sets AR lower by 0.5 it should lower each map AR by this value.

**Comment by Stoppedpuma** [comment]
I'm a bit confused by this, for example setting AR to 9.5 in Difficulty Adjust should set all maps to AR 9.5. Are you asking for an option to lower based on the original AR such as 8.6 -> 8.1, 9.8 -> 9.3?

Alternatively if you mean having DA keep a specific AR value when using rate change mods then see #15249.

**Reply by adriankiezik** [reply]
Exactly that, for example if CS is set higher by 0.5 and locked, the values should be:

Original CS -> Adjusted CS
3.5 -> 4.0
4.2 -> 4.7

Same for setting it lower. Sorry for confusion. Also, I believe this thing exists in McOsu for reference.

**Reply by Stoppedpuma** [reply]
Doesn't the lock feature in McOsu just ignore overrides from rate changes? I've never used it for what you're requesting.

**Reply by adriankiezik** [reply]
Oh, you're right, it doesn't work this way. I guess I suffer from dementia. Anyways, I think it's cool idea to have something like this in the game
